### PR TITLE
[BEAM-1611] Explicitly GBK before stateful ParDo in Dataflow batch

### PR DESCRIPTION
**Pull request by kennknowles**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by kennknowles** [comment]
Managed a service-side fix; it will require a worker update so leaving the JIRA open.

**Comment by kennknowles** [comment]
Run Dataflow RunnableOnService

**Comment by kennknowles** [comment]
Run Dataflow RunnableOnService

**Comment by kennknowles** [comment]
R: @tgroh 
CC: @aaltay for release FYI. The test has been run offline; we are confident in this solution. This is easy to cherry pick.

**Comment by kennknowles** [comment]
Run Dataflow RunnableOnService

**Comment by kennknowles** [comment]
R: @bjchambers WDYT?

**Comment by kennknowles** [comment]
Filed [BEAM-1647](https://issues.apache.org/jira/browse/BEAM-1647) about the RunnableOnService suite timing out. I checked the logs and most of the tests made it under the limit (we are just slightly over), in particular `ParDoTest` succeeded.

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/BatchStatefulParDoOverrides.java [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+ * {@link PTransformOverrideFactory PTransformOverrideFactories} that expands to correctly implement
+ * stateful {@link ParDo} using window-unaware {@link GroupByKeyAndSortValuesOnly} to linearize
+ * processing per key.
+ *
+ * <p>This implementation relies on implementation details of the Dataflow runner, specifically
+ * standard fusion behavior of {@link ParDo} tranforms following a {@link GroupByKey}.
+ */
+public class BatchStatefulParDoOverrides {
+
+  public static <K, InputT, OutputT>
```
Comment (bjchambers) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/BatchStatefulParDoOverrides.java]: Javadoc public methods here and below?

**Pull request merged**